setup.py

```diff
@@ -177,7 +188,109 @@ def build_extension(self, ext):
+                        NANOARROW_ARROW_ITERATOR_SRC_DIR, "CArrowIterator.cpp"
+                    ),
+                    os.path.join(
+                        NANOARROW_ARROW_ITERATOR_SRC_DIR, "CArrowChunkIterator.cpp"
+                    ),
+                    os.path.join(
+                        NANOARROW_ARROW_ITERATOR_SRC_DIR, "CArrowTableIterator.cpp"
+                    ),
+                    os.path.join(NANOARROW_ARROW_ITERATOR_SRC_DIR, "SnowflakeType.cpp"),
+                    os.path.join(
+                        NANOARROW_ARROW_ITERATOR_SRC_DIR, "BinaryConverter.cpp"
+                    ),
+                    os.path.join(
+                        NANOARROW_ARROW_ITERATOR_SRC_DIR, "BooleanConverter.cpp"
+                    ),
+                    os.path.join(
+                        NANOARROW_ARROW_ITERATOR_SRC_DIR, "DecimalConverter.cpp"
+                    ),
+                    os.path.join(NANOARROW_ARROW_ITERATOR_SRC_DIR, "DateConverter.cpp"),
+                    os.path.join(
+                        NANOARROW_ARROW_ITERATOR_SRC_DIR, "FloatConverter.cpp"
+                    ),
+                    os.path.join(NANOARROW_ARROW_ITERATOR_SRC_DIR, "IntConverter.cpp"),
+                    os.path.join(
+                        NANOARROW_ARROW_ITERATOR_SRC_DIR, "StringConverter.cpp"
+                    ),
+                    os.path.join(NANOARROW_ARROW_ITERATOR_SRC_DIR, "TimeConverter.cpp"),
+                    os.path.join(
+                        NANOARROW_ARROW_ITERATOR_SRC_DIR, "TimeStampConverter.cpp"
+                    ),
+                    os.path.join(
+                        NANOARROW_ARROW_ITERATOR_SRC_DIR, "Python", "Common.cpp"
+                    ),
+                    os.path.join(
+                        NANOARROW_ARROW_ITERATOR_SRC_DIR, "Python", "Helpers.cpp"
+                    ),
+                    os.path.join(NANOARROW_ARROW_ITERATOR_SRC_DIR, "Util", "time.cpp"),
+                    NANOARROW_LOGGING_SRC_DIR + "/logging.cpp",
+                    os.path.join(NANOARROW_ARROW_ITERATOR_SRC_DIR, "nanoarrow.c"),
+                    os.path.join(NANOARROW_ARROW_ITERATOR_SRC_DIR, "nanoarrow_ipc.c"),
+                    os.path.join(NANOARROW_ARROW_ITERATOR_SRC_DIR, "flatcc.c"),
+                ]
+                ext.include_dirs.append(NANOARROW_ARROW_ITERATOR_SRC_DIR)
+                ext.include_dirs.append(NANOARROW_LOGGING_SRC_DIR)
+
+                if sys.platform == "win32":
+                    if not any("/std" not in s for s in ext.extra_compile_args):
+                        ext.extra_compile_args.append("/std:c++17")
+                elif sys.platform == "linux" or sys.platform == "darwin":
+                    if "std=" not in os.environ.get("CXXFLAGS", ""):
+                        ext.extra_compile_args.append("-std=c++17")
+                        ext.extra_compile_args.append("-D_GLIBCXX_USE_CXX11_ABI=0")
+                    if sys.platform == "darwin":
```

Comment:
@xhochy thanks for the suggestions.
I'm not familiar with the mac compiler, so have some dumb questions:
- does specifying `macosx-version-min` will force the connector to be compiled with the specific min version, or it's >=
- are multiple compiler version allowed on mac?

Reply:
> does specifying macosx-version-min will force the connector to be compiled with the specific min version, or it's >=

The use case would be a higher version. Right now the whole stack is actually moving towards this as LLVM 17 requires `10.13` instead of previously `10.9`. So this won't be relevant right now but I would expect in future to see this increase even more in cases where you bring in newer tools in the toolchain.

> are multiple compiler version allowed on mac?

Yes, as part of conda-forge, we ship and build with a newer compiler than that shipped with macOS/Xcode. This could be clang/LLVM 17 in about 1-2 weeks. That release is roughly 1-2 weeks now and probably needs quite some while until it ships officially via Apple Channels.

Reply:
fixed in the #1758